Source/JavaScriptCore/runtime/FunctionPrototype.cpp

```diff
@@ -108,6 +109,8 @@ static ALWAYS_INLINE bool isAllowedReceiverFunctionForCallerAndArguments(JSFunct
         return false;
 
     FunctionExecutable* executable = function->jsExecutable();
+    if (executable->implementationVisibility() == ImplementationVisibility::Private)
```

Comment:
Can this check even pass after

```
if (function->isHostOrBuiltinFunction())
         return false;
```

?

Maybe we should remove it, as builtin functions can't be receivers for `caller`?

Reply:
I think we should first check this thing since we have builtins with `::Private`. And in that case, we should not stop stack-walking here, instead, we should continue by skipping it.

Reply:
Yes, non-builtin function can be Private (e.g. class static initializer function).